### ISSUE TITLE
Getting error when decoding event with indexed fields.

### ISSUE THREAD

**Issue by coinsandsteeldev** [ethjs/ethjs#30]
Hi! I'm using Ganache 1.0.1 and ethjs 0.4.0.
Here how i use watcher: 
  ```
this.FightFinished = Contracts.zoneContract.FightFinished(
    { toBlock: 'latest', player: Client.activeAccount }
)
  this.FightFinished.new()
  this.FightFinished.watch(r => {  
  })
```
I believe this was outlined at [https://github.com/ethjs/ethjs-abi/issues/15](url)
But no response at all, i'll try get response here instead. 

Log from watch:

`Error: [ethjs-filter] while decoding filter change event data from RPC '[{"logIndex":"2","transactionIndex":"0","transactionHash":"0x2823403d01f35e4d04a2316c72c9a9906160cafa447099644c6af0c48c18261e","blockHash":"0x745dd38108efe3469825dcbc9986a27717ccee25c95cae8cb338886eb73aa639","blockNumber":"92","address":"0xb57e97e510c60673412307547142e7492cd2cb1f","data":"0x0000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000002","topics":["0x79b7b95e80b1a19a31c1a9cdfdbc01f323ac30aef005dd716b3474f8d4b72c68","0x000000000000000000000000e91d5216f69648ce919442e2c426cf48738f4101"],"type":"mined"}]': TypeError: Cannot read property 'slice' of undefined`

**Comment by SilentCicero** [comment]
Okay, so it's possible events here have been altered.

I'll take a deeper look tomorrow. Thanks!

**Comment by coinsandsteeldev** [comment]
Thanks, would be awesome, this problem doesn't allow to continue some parts of development.

**Comment by SilentCicero** [comment]
Yeah, very strange. And have you tried just polling eth getLogs and decoding the results with ethjs abi?

**Comment by coinsandsteeldev** [comment]
No, i will try

**Comment by coinsandsteeldev** [comment]
Yes, it worked!

**Comment by SilentCicero** [comment]
Amazing! I should really make that an easier method for all this

**Comment by coinsandsteeldev** [comment]
That is great, but how to listen for new Logs?

**Comment by coinsandsteeldev** [comment]
@SilentCicero

**Comment by SilentCicero** [comment]
So, I usually just run a polling loop.

Checkout ethjs-extras (a rouge library I wrote) for more details.

NPM:
https://www.npmjs.com/package/ethjs-extras

Github:
https://github.com/SilentCicero/ethjs-extras

Contract Code:
```

const {
  Eth,
  EthContract,
  onAccount,
  onReceipt,
  onBlock,
  call,
} = require('ethjs-extras'); // or import..


// setup a light contract object, use solidity spec and or abi spec. const
simpleStore = new EthContract({ network: 'rinkeby', address: '
0x01a528451419d562a6752318a6fb71898fa5d840', methods: [ 'SimpleStore(address
_master)', 'set()', 'Set(uint256 _value)', 'master():(address)', '
get():(uint256)', ], }); // contract methods, promise only simpleStore.get
().then(console.log).catch(console.log); // using shorthand notation
simpleStore.master.call().then(console.log).catch(console.log); // using
.call() notation simpleStore.onEvent(console.log); // polls for events,
receives, decodes and fires callback <-------------------------- simpleStore
.getLogs({ fromBlock: 0 }).then(console.log).catch(console.log);


```

Event Polling I do here (every block):
https://github.com/SilentCicero/ethjs-extras/blob/master/src/index.js#L190


hope this helps, sorry swamped with work!

On Wed, May 16, 2018 at 2:49 AM, cryptodungeonsdev <notifications@github.com
